### PR TITLE
[WIP] Added a sample for file-based instance mapping.

### PR DESCRIPTION
**Pull request by SzymonPobiega**
I'll keep adding routing samples until you get tired reviewing them ;-)

I marked it as WIP because it requires fixing [a bug](https://github.com/Particular/NServiceBus/pull/3867) in the core (in the file instance mapping feature)

### CODE REVIEW

**Comment by SimonCropp** [comment]
@SzymonPobiega i did a cleanup. see the commits above

**Comment by SzymonPobiega** [comment]
@SimonCropp thanks. I like it. Forgot to remove NH again :\

**Comment by SzymonPobiega** [comment]
Does it mean that it *only* has to pass through @weralabaj and will be good to go?

**Comment by weralabaj** [comment]
should be good with just Simon's review

File: samples/routing/instance-mapping-file/sample.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: Mapping endpoint instances using a shared file
+summary: Mapping logical endpoints to physical instances using a shared file
+component: Core
+tags:
+ - Routing
+ - MSMQ
+---
+
+The sample demonstrates to use a file to describe the mapping between logical endpoints and their physical instances. This is important for store-and-forward transports such as MSMQ because the physical topology can differ from the logical one.
```
Comment (weralabaj) [inline samples/routing/instance-mapping-file/sample.md]: > The sample demonstrates to use a

**how** to use a

Comment (weralabaj) [inline samples/routing/instance-mapping-file/sample.md]: > This is important for store-and-forward transports such as MSMQ because the physical topology can differ from the logical one.

Not sure if that's obvious what you mean, maybe some example? Is that that logically you have a single endpoint, but physically there might be multiple instances? If so, mention it.

Comment (SzymonPobiega) [inline samples/routing/instance-mapping-file/sample.md]: I removed that sentence.

**Comment by SzymonPobiega** [comment]
@weralabaj corrected. Please hold on with the merge until that bug is done and released. Otherwise the sample does not work :(

**Comment by indualagarsamy** [comment]
hey @SzymonPobiega - any updates on this?

**Comment by weralabaj** [comment]
Status update: The PR with bug fix was merged, but we need to wait for new core beta release. Otherwise users will experience exceptions after download.

TL;DR: To be tested and released after new core beta release (0006).
